yt/data_objects/selection_objects/data_selection_objects.py

```diff
@@ -214,7 +214,8 @@ def get_data(self, fields=None):
 
         for f, v in read_particles.items():
             self.field_data[f] = self.ds.arr(v, units=finfos[f].units)
-            self.field_data[f].convert_to_units(finfos[f].output_units)
+            if finfos[f].units != finfos[f].output_units:
+                self.field_data[f].convert_to_units(finfos[f].output_units)
```

Comment:
note to reviewers: this change is necessary because unit conversions on unyt arrays always cast the result to float64. e.g., `unyt.unyt_array([1, 2], "1", dtype='int').to("1")` yields an array of type `float64`. So only converting if the units differ allow the int fields to persist through the data read.